fix incorrect namespacing applied to Redis hash field keys

2ykwang opened the pull request:
This changes addresses an issue where namespacing was applied to the field instead of the key.


**behavior:**
- before: Namespacing was applied to the field parameter (e.g, hash key) when it should have been applied to the key parameter.
- after: Namespacing is correctly applied to the key parameter, leaving the field name unaltered.

2ykwang committed:
restrict namespacing to hash name only

2ykwang commented:
should I look into something specific about the GitHub CI failure? @WisdomPill

WisdomPill commented:
hello @2ykwang,

thanks for your contribution! 

so only changelog and ruff are left to fix:
- changelog needs a file called `changelog.d/765.bugfix` because this PR is number 765 and its reason is a bug fix, inside the file write a description of what the PR is fixing, this will later be put into the changelog when releasing automatically with `towncrier`
- as for ruff you can remove `ANN101` and add `A004` [here](https://github.com/jazzband/django-redis/blob/ed279990856373f59dc7d76644d4a02426d4eca8/.ruff.toml#L116) and it will solve itself

2ykwang committed:
Add changelog fix

2ykwang commented:
I’ve added the changelog  @WisdomPill

WisdomPill reviewed (changes requested):
LGTM, I wonder how tests didn't break. Could you try using different version and checking the key in the tests?

Also please fix ruff's error

WisdomPill commented:
also FYI, there's no need to mention me in a comment, just ask for a review, I will see the notifications and I will be happy to review 😄

2ykwang commented:
> LGTM, I wonder how tests didn't break. Could you try using different version and checking the key in the tests?

because `hset`, `hexists`, and all other methods that require keys use the same namespacing logic via make_key, ensuring consistent key generation.
and there are no test cases that specifically verify the namespacing of hash keys.

2ykwang committed:
Fix ruff lint ignore configuration

Updated the ruff lint.ignore settings to include inline documentation links for each ignored rule.
Also added "A004", "A005"

terencehonles commented:
This has been merged via #812
